set.mm

```diff
@@ -633662,6 +633662,96 @@ deduction version (Contributed by metakunt, 22-May-2024.) $)
+      ( wfn cin c0 wceq wcel wa w3a cun cfv jca 3jca fvun2 syl ) ADBKZECKZBCLMN
+      ZFCOZPZQFDERSFESNAUDUEUHGHAUFUGIJTUABCDEFUBUC $.
+  $}
+
+  ${
+    fnund.1 $e |- ( ph -> F Fn A ) $.
+    fnund.2 $e |- ( ph -> G Fn B ) $.
+    fnund.3 $e |- ( ph -> ( A i^i B ) = (/) ) $.
+    $( The union of two functions with disjoint domains, a deduction version.
+       (Contributed by metakunt, 28-May-2024.) $)
+    fnund $p |- ( ph -> ( F u. G ) Fn ( A u. B ) ) $=
+      ( wfn wa cin c0 wceq cun jca31 fnun syl ) ADBIZECIZJBCKLMZJDENBCNIARSTFGH
+      OBCDEPQ $.
+  $}
+
+  ${
+    fzsplitnd.1 $e |- ( ph -> K e. ( M ... N ) ) $.
+    $( Split a finite interval of integers into two parts.  (Contributed by
+       metakunt, 28-May-2024.) $)
+    fzsplitnd $p |- ( ph -> ( M ... N ) =
+     ( ( M ... ( K - 1 ) ) u. ( K ... N ) ) ) $=
+      ( cfz co c1 cmin cun cuz cfv wcel wceq elfzuz syl elfzelzd mpbird syl2anc
+      caddc zcnd 1cnd npcand eleq1d 1zzd zsubcld elfzuz3 fveq2d eleq2d fzsplit2
+      cz peano2uzr oveq1d uneq2d eqtrd ) ACDFGZCBHIGZFGZUQHTGZDFGZJZURBDFGZJAUS
+      CKLZMZDUQKLMZUPVANAVDBVCMZABUPMZVFEBCDOPAUSBVCABHABABCDEQZUAAUBUCZUDRAUQU
+      KMDUSKLZMZVEABHVHAUEUFAVKDBKLZMZAVGVMEBCDUGPAVJVLDAUSBKVIUHUIRUQDULSUQCDU
+      JSAUTVBURAUSBDFVIUMUNUO $.
+  $}
+
+  ${
+    $d K x $.  $d M x $.  $d N x $.  $d ph x $.
+    fzsplitnr.1 $e |- ( ph -> M e. ZZ ) $.
+    fzsplitnr.2 $e |- ( ph -> N e. ZZ ) $.
+    fzsplitnr.3 $e |- ( ph -> K e. ZZ ) $.
+    fzsplitnr.4 $e |- ( ph -> M <_ K ) $.
+    fzsplitnr.5 $e |- ( ph -> K <_ N ) $.
+    $( Split a finite interval of integers into two parts.  (Contributed by
+       metakunt, 28-May-2024.) $)
+    fzsplitnr $p |- ( ph -> ( M ... N ) =
+     ( ( M ... ( K - 1 ) ) u. ( K ... N ) ) ) $=
+      ( co c1 wcel wa wbr cle zred adantr cz adantl elfzd vx cfz cmin cun cv wo
+      clt elfzelz lelttric syl2anc orcomd zltlem1 ad2antrr 1zzd zsubcld elfzle1
+      cr wb simpr elfzle2 impbida bitrd bicomd orbi12d mpbird 1red resubcld cc0
+      0le1 subge02d mpbii letrd elun2 elun wi a1i sylan2b sylan2 jaodan syl6bbr
+      jaod imp eqrdv ) AUACDUBJZCBKUCJZUBJZBDUBJZUDZAUAUEZWDLZWIWFLZWIWGLZUFZWI
+      WHLZAWJWMAWJMZWMWIBUGNZBWIONZUFWOWQWPWOBUQLZWIUQLWQWPUFAWRWJABGPZQWOWIWJW
+      IRLZAWICDUHSZPBWIUIUJUKWOWKWPWLWQWOWPWKWOWPWIWEONZWKWOWTBRLZWPXBURXAAXCWJ
+      GQZWIBULUJWOXBWKWOXBMZWICWEACRLZWJXBEUMXEBKWOXCXBXDQXEUNUOWOWTXBXAQWOCWIO
+      NZXBWJXGAWICDUPSQWOXBUSTWKXBWOWICWEUTZSVAVBVCWOWLWQWLWQWOWIBDUPZSWOWQMWIB
+      DWOXCWQXDQADRLZWJWQFUMWOWTWQXAQWOWQUSWOWIDONZWQWJXKAWICDUTSQTVAVDVEAWKWJW
+      LAWKMZWICDAXFWKEQAXJWKFQZWKWTAWICWEUHZSZWKXGAWICWEUPSXLWIWEDXLWIXOPXLBKAW
+      RWKWSQZXLVFVGZXLDXMPZWKXBAXHSXLWEBDXQXPXRAWEBONZWKAVHKONXSVIABKWSAVFVJVKQ
+      ABDONWKIQVLVLTAWLMZWICDAXFWLEQZAXJWLFQWLAWNWTWIWGWFVMWNAWMWTWIWFWGVNZAWMW
+      TAWKWTWLWKWTVOAXNVPWLWTVOAWIBDUHVPWAWBVQVRZXTCBWIXTCYAPAWRWLWSQXTWIYCPACB
+      ONWLHQWLWQAXISVLWLXKAWIBDUTSTVSVAYBVTWC $.
+  $}
```

Comment:
Why is this proof so long? It should be much shorter using ~fzsplitnd.

Reply:
It's because my brain went afk. Jokes aside, I tried to prove a different statement with weaker hypotheses, which would have been nicer to use. Turns out, the statement I've tried to prove was a contradiction, so I couldn't obviously prove it. I had to strengthen the hypotheses which sadly made the theorem that uses this statement ungodly ugly. But it worked and I've completely forgot that the thing I have just proved is completely and utterly useless. Well actually it is useful because I do need to use it, but I wished I could have used my wishful wrong version.

So the wrong theorem was something originally like this.
```
    fzsplitnr.1 $e |- ( ph -> M e. ZZ ) $.
    fzsplitnr.2 $e |- ( ph -> N e. ZZ ) $.
    fzsplitnr.3 $e |- ( ph -> K e. ZZ ) $.
    fzsplitnr.4 $e |- ( ph -> A e. ZZ ) $.
    fzsplitnr.5 $e |- ( ph -> B e. ZZ ) $.
    fzsplitnr.6 $e |- ( ph -> M <_ A ) $.
    fzsplitnr.7 $e |- ( ph -> B <_ N ) $.


    fzsplitnr $p |- ( ph -> ( A ...B ) =
     ( ( A ... ( K - 1 ) ) u. ( K ... B ) ) ) $=
```
Yeah that clearly is wrong, but it would have saved me a lot of steps.

Nonetheless, I have fixed the theorem, it is indeed one line.